packages/devtools_app/lib/src/flutter/common_widgets.dart

```diff
@@ -113,3 +116,163 @@ class DefaultTaggedText extends StatelessWidget {
+  ///
+  /// [firstChild] will be placed before [secondChild] in a [Row].
+  const Split.horizontal({
+    Key key,
+    @required Widget firstChild,
+    @required Widget secondChild,
+    double initialFirstFraction,
+  }) : this._(
+          key,
+          Axis.horizontal,
+          firstChild,
+          secondChild,
+          initialFirstFraction,
+        );
+
+  /// Builds a [Split] with [Axis.vertical] direction.
+  ///
+  /// [firstChild] will be placed before [secondChild] in a [Column].
+  const Split.vertical({
+    Key key,
+    @required Widget firstChild,
+    @required Widget secondChild,
+    double initialFirstFraction,
+  }) : this._(
+          key,
+          Axis.vertical,
+          firstChild,
+          secondChild,
+          initialFirstFraction,
+        );
+
+  const Split._(
+    Key key,
+    this.axis,
+    this.firstChild,
+    this.secondChild,
+    double initialFirstFraction,
+  )   : initialFirstFraction = initialFirstFraction ?? 0.5,
+        super(key: key);
+
+  /// The main axis the children will lay out on.
+  ///
+  /// If [Axis.horizontal], the children will be placed in a [Row]
+  /// and they will be horizontally resizable.
+  ///
+  /// If [Axis.vertical], the children will be placed in a [Column]
+  /// and they will be vertically resizable.
+  ///
+  /// Cannot be null.
+  final Axis axis;
+
+  /// The child that will be laid out first along [axis].
+  final Widget firstChild;
+
+  /// The child that will be laid out last along [axis].
+  final Widget secondChild;
+
+  /// The fraction of the layout to allocate to [firstChild].
+  ///
+  /// [secondChild] will receive a fraction of `1 - initialFirstFraction`.
+  final double initialFirstFraction;
+
+  /// The key passed to the divider between [firstChild] and [secondChild].
+  ///
+  /// Visible to grab it in tests.
+  @visibleForTesting
+  Key get dividerKey => Key('$this dividerKey');
+
+  /// The size of the divider between [firstChild] and [secondChild] in
+  /// logical pixels (dp, not px).
+  static const double dividerMainAxisSize = 10.0;
+
+  @override
+  State<StatefulWidget> createState() => _SplitState();
+}
+
+class _SplitState extends State<Split> {
+  double firstFraction;
+  double get secondFraction => 1 - firstFraction;
+  bool get isHorizontal => widget.axis == Axis.horizontal;
+
+  @override
+  void initState() {
+    super.initState();
+    firstFraction = widget.initialFirstFraction;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(builder: _buildLayout);
+  }
+
+  Widget _buildLayout(BuildContext context, BoxConstraints constraints) {
+    final width = constraints.maxWidth;
+    final height = constraints.maxHeight;
+    final axisSize = isHorizontal ? width : height;
+    const halfDivider = Split.dividerMainAxisSize / 2.0;
+    // The fraction of the layout the divider needs to take up from each child.
+    final halfDividerFraction = halfDivider / axisSize;
+
```

Comment:
Why are we using fractions instead of just subtracting Split.dividerMainAxisSize / 2.0?

Reply:
Good suggestion, I've removed this field.